src/app/parcels/ActionBar/Statuses.tsx

```diff
@@ -36,24 +38,28 @@ export const saveParcelStatus = async (
     parcelIds: string[],
     statusName: StatusType,
     statusEventData?: string | null,
+    clientIds?: string[],
+    action?: string,
     date?: Dayjs
 ): Promise<SaveParcelStatusResult> => {
     const timestamp = (date ?? dayjs()).toISOString();
     const eventsToInsert = parcelIds
-        .map((parcelId: string) => {
+        .map((parcelId: string, index) => {
             return {
                 new_parcel_status: statusName,
                 parcel_id: parcelId,
                 event_data: statusEventData,
+                client_id: clientIds?.slice()[index],
```

Comment:
Really I just want `clientIds[index]`. But since `clientIds` is `string[] | undefined`, I needed the `?`. I don't think that typescript likes the `[index]` right after the `?` so `.slice()` helps with typescript understanding what I want. Is there a better way to do this?

Reply:
You can use `clientIds?.at(index)` I think